src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CachePoolPass.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\DefinitionDecorator;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class CachePoolPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        foreach ($container->findTaggedServiceIds('cache.pool') as $id => $tags) {
+            $adapter = $pool = $container->getDefinition($id);
+            $tags[0] += array('namespace' => $this->getNamespace($id));
+
+            while ($adapter instanceof DefinitionDecorator) {
+                $adapter = $container->findDefinition($adapter->getParent());
+                if ($t = $adapter->getTag('cache.pool')) {
+                    $tags[0] += $t[0];
+                }
+            }
+            if (!isset($tags[0]['namespace_arg_index'])) {
+                throw new \InvalidArgumentException(sprintf('Invalid "cache.pool" tag for service "%s": attribute "namespace_arg_index" is missing.', $id));
+            }
+            if ($pool->isAbstract()) {
+                continue;
+            }
+            if (isset($tags[0]['provider_service']) && is_string($tags[0]['provider_service'])) {
+                $tags[0]['provider_service'] = new Reference($tags[0]['provider_service']);
+            }
+            foreach ($tags[0] as $argName => $argValue) {
+                if (isset($tags[0][$argName.'_arg_index']) && 0 <= $tags[0][$argName.'_arg_index']) {
+                    $pool->replaceArgument($tags[0][$argName.'_arg_index'], $argValue);
+                }
+            }
+        }
+    }
+
+    private function getNamespace($id)
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid variables with short names like $id. Configured minimum length is 3.](https://www.codacy.com/app/git_9/symfony/file/2331802099/issues/source?bid=3207821&fileBranchId=3207834#l56)